### PR TITLE
[#10920] Add front-end tests for Instructor Session Page

### PR DESCRIPTION
**Pull request by tsy24**
Part of #10920 

**Outline of Solution**
Added unit tests for Instructor Sessions Page.

### CODE REVIEW

File: src/web/app/pages-instructor/instructor-sessions-page/instructor-sessions-page.component.spec.ts [hunk 1]
```diff
@@ -1,31 +1,433 @@
...
+      .mockReturnValue(of(testFeedbackSession3));
+
+    component.restoreRecycleBinFeedbackSession(recycleBinFeedbackSessionRowModel1);
+    expect(sessionSpy).toHaveBeenCalledTimes(1);
+    expect(sessionSpy).toHaveBeenLastCalledWith('CS1231', 'Third Session');
+    expect(component.sessionsTableRowModels.length).toEqual(1);
+    expect(component.recycleBinFeedbackSessionRowModels.length).toEqual(0);
+  });
+
+  it('should restore all session', () => {
```
Comment (zhaojj2209) [inline src/web/app/pages-instructor/instructor-sessions-page/instructor-sessions-page.component.spec.ts:268]:
```suggestion
  it('should restore all sessions', () => {
```

File: src/web/app/pages-instructor/instructor-sessions-page/instructor-sessions-page.component.spec.ts [hunk 2]
```diff
@@ -1,31 +1,433 @@
...
+    jest.spyOn(ngbModal, 'open').mockReturnValue(mockModalRef);
+
+    component.permanentDeleteSession(recycleBinFeedbackSessionRowModel1);
+    await promise;
+
+    expect(ngbModal.open).toHaveBeenCalledWith(SessionPermanentDeletionConfirmModalComponent);
+    expect(sessionSpy).toHaveBeenCalledTimes(1);
+    expect(sessionSpy).toHaveBeenLastCalledWith('CS1231', 'Third Session');
+
+    // expect(component.activeCourses.length).toEqual(0);
```
Comment (zhaojj2209) [inline src/web/app/pages-instructor/instructor-sessions-page/instructor-sessions-page.component.spec.ts:313]: Remove comment

**Review by zhaojj2209 — COMMENTED** [review]
Good coverage! Just some nits

**Comment by fsgmhoward** [comment]
@tsy24 After you fix the commented parts please mark it as "resolved" or post a follow up comment there! Thanks.

**Review by zhaojj2209 — APPROVED** [review]
LGTM, thanks for the changes!

**Review by samuelfangjw — APPROVED** [review]

**Pull request merged**